### ISSUE TITLE
Semaphore must not have any pending operations

### ISSUE THREAD

**Issue by hecrj** [gfx-rs/wgpu#5644]
**Description**
I have updated `iced` to use the latest release (see https://github.com/iced-rs/iced/pull/2417). While `0.19` works perfectly fine, `0.20` is very sluggish and I seem to be constantly hitting a validation error:

```
2024-05-01T14:25:29.145657Z ERROR wgpu_hal::vulkan::instance: VALIDATION [VUID-vkAcquireNextImageKHR-semaphore-01779 (0x5717e75b)]
        Validation Error: [ VUID-vkAcquireNextImageKHR-semaphore-01779 ] Object 0: handle = 0x4fac1c0000000032, type = VK_OBJECT_TYPE_SEMAPHORE; | MessageID = 0x5717e75b | vkAcquireNextImageKHR():  Semaphore must not have any pending operations. The Vulkan spec states: If semaphore is not VK_NULL_HANDLE it must not have any uncompleted signal or wait operations pending (https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/vkspec.html#VUID-vkAcquireNextImageKHR-semaphore-01779)
2024-05-01T14:25:29.145682Z ERROR wgpu_hal::vulkan::instance:   objects: (type: SEMAPHORE, hndl: 0x4fac1c0000000032, name: ?)
```

The upgrade was straightforward—it only involved adding default `compilation_options` everywhere, so not quite sure what's up.

**Repro steps**
Most examples of the `iced` repository (like the `tour`) in the `wgpu-0.20` branch have the issue.

**Expected vs observed behavior**
It should work well like `0.19` and not produce validation errors.

**Extra materials**
Validation log pasted in description.

**Platform**
Arch Linux, X11, NVIDIA RTX 4090, proprietary drivers.

**Comment by cwfitzgerald** [comment]
This is almost certainly https://github.com/gfx-rs/wgpu/issues/5559

**Comment by hecrj** [comment]
@cwfitzgerald Was this introduced in `0.20`? `0.19` does not have the issue I reported.

I ran `cargo flamegraph` with one of the examples and noticed that almost all of the time is now spent in `Surface::present`:

![flamegraph](https://github.com/gfx-rs/wgpu/assets/518289/f1c0b832-0690-4c85-8cfa-42b9212face2)

**Comment by hecrj** [comment]
Here is `0.19` with the same example, for comparison:

![flamegraph](https://github.com/gfx-rs/wgpu/assets/518289/48a90d32-e7ef-4279-8a3e-671c7b0227c9)

**Comment by hecrj** [comment]
Same behavior with the `cube` example of the repository:

# `v0.20`

![flamegraph](https://github.com/gfx-rs/wgpu/assets/518289/5cc9df20-f507-46d1-9fda-e11fb5b4a4e5)

# `v0.19`

![flamegraph](https://github.com/gfx-rs/wgpu/assets/518289/9cf105f1-e655-4356-8570-b7c6454e9ced)

**Comment by cwfitzgerald** [comment]
I'm honestly not sure when synchronization changed, I just noticed it's very wrong on trunk a bit ago. It seems to be causing problems, so I'll try and fix this.

**Comment by cwfitzgerald** [comment]
This was closed as a dupe of #5559

**Comment by cwfitzgerald** [comment]
@hecrj Can you try #5681 to see if this fixes your issue?
